secure-api-gateway-ob-uk-rs-server/src/main/java/com/forgerock/sapi/gateway/ob/uk/rs/server/api/obie/payment/v4_0_0/domesticscheduledpayments/DomesticScheduledPaymentConsentsApiController.java

```diff
@@ -0,0 +1,116 @@
+import com.forgerock.sapi.gateway.ob.uk.common.error.OBErrorResponseException;
+import com.forgerock.sapi.gateway.ob.uk.rs.obie.api.payment.v4_0_0.domesticscheduledpayments.DomesticScheduledPaymentConsentsApi;
+import com.forgerock.sapi.gateway.ob.uk.rs.server.api.obie.payment.factories.v4_0_0.OBWriteDomesticScheduledConsentResponse5Factory;
+import com.forgerock.sapi.gateway.ob.uk.rs.validation.obie.OBValidationService;
+import com.forgerock.sapi.gateway.rcs.consent.store.client.payment.domesticscheduled.DomesticScheduledPaymentConsentStoreClient;
+import com.forgerock.sapi.gateway.rcs.consent.store.datamodel.payment.domesticscheduled.v3_1_10.CreateDomesticScheduledPaymentConsentRequest;
+import com.forgerock.sapi.gateway.rcs.consent.store.datamodel.payment.domesticscheduled.v3_1_10.DomesticScheduledPaymentConsent;
+
+import jakarta.servlet.http.HttpServletRequest;
+import uk.org.openbanking.datamodel.v4.payment.OBWriteDomesticScheduledConsent4;
+import uk.org.openbanking.datamodel.v4.payment.OBWriteDomesticScheduledConsentResponse5;
+
+@Controller("DomesticScheduledPaymentConsentsApiV4.0.0")
+public class DomesticScheduledPaymentConsentsApiController implements DomesticScheduledPaymentConsentsApi {
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    private final DomesticScheduledPaymentConsentStoreClient consentStoreApiClient;
+
+    private final OBValidationService<OBWriteDomesticScheduledConsent4> consentValidator;
+
+    private final OBWriteDomesticScheduledConsentResponse5Factory consentResponseFactory;
+
+    public DomesticScheduledPaymentConsentsApiController(
+            @Qualifier("v4.0.0RestDomesticScheduledPaymentConsentStoreClient") DomesticScheduledPaymentConsentStoreClient consentStoreApiClient,
+            OBValidationService<OBWriteDomesticScheduledConsent4> consentValidator,
+            OBWriteDomesticScheduledConsentResponse5Factory consentResponseFactory) {
+
+        this.consentStoreApiClient = consentStoreApiClient;
+        this.consentValidator = consentValidator;
+        this.consentResponseFactory = consentResponseFactory;
+    }
+
+    @Override
+    public ResponseEntity<OBWriteDomesticScheduledConsentResponse5> createDomesticScheduledPaymentConsents(
+            String authorization,
+            String xIdempotencyKey,
+            String xJwsSignature,
+            OBWriteDomesticScheduledConsent4 obWriteDomesticScheduledConsent4,
+            String xFapiAuthDate,
+            String xFapiCustomerIpAddress,
+            String xFapiInteractionId,
+            String xCustomerUserAgent,
+            String apiClientId,
+            HttpServletRequest request,
+            Principal principal) throws OBErrorResponseException {
+
+        logger.info("Processing createDomesticScheduledPaymentConsents request - consent: {}, idempotencyKey: {}, apiClient: {}, x-fapi-interaction-id: {}",
```

Comment:
Expect this should be trace level?

Reply:
And line 90... Maybe also elsewhere.

Too much logging causes noise and can become expensive. These both look like trace-level logs.

Reply:
Also, what is an `idempotencyKey` - if it's a security token then it should probably not be logged at all.

Reply:
As we only need general info on the request and not step-by-step tracing in here info level is fine.
An `idempotency key` is a unique string used to protect against accidental duplicate calls. 
Basically if we call an API multiple times with the same request, it's considered to be `idempotent` if it has the same effect on the server as if we had only sent one request.

Reply:
That doesn't sound right regarding info log level. The customer expects to see this info for every request? 

I'd expect they'd only want it in a situation where the system was producing unexpected result. If that's the case then debug or trace is right, to avoid filling the logs with mostly meaningless (and expensive) messages.

Reply:
Changed log level. :)